docs/api/serviceregistry/device-discovery-generic-http.md

```diff
@@ -1,25 +1,205 @@
-# device-discovery GENERIC-HTTP
+# device-discovery IDD
+**GENERIC-HTTP & GENERIC-HTTPS**
 
 ## Overview
 
-TODO
+This page describes the GENERIC-HTTP and GENERIC-HTTPS service interface of device-discovery, which enables both
+application and core/support systems to lookup, register and revoke devices on which the Local Cloud’s systems
+are running. Device representation is not necessary for the base functionalities of a Local Cloud but in certain
+use cases (e.g. enabling onboarding) is needed. It’s implemented using protocol, encoding as stated in the
+following tables:
 
-## Interface Description
+**GENERIC-HTTP**
+
+Profile type | type | Version
+--- | --- | ---
+Transfer protocol | HTTP | 1.1
+Data encryption | N/A | -
+Encoding | JSON | RFC 8259
+Compression | N/A | -
+
+**GENERIC-HTTPS**
+
+Profile type | type | Version
+--- | --- | ---
+Transfer protocol | HTTPS | 1.1
+Data encryption | TLS | -
+Encoding | JSON | RFC 8259
+Compression | N/A | -
+
+Hereby the **Interface Design Description** (IDD) is provided to the [device-discovery – Service Description](../../assets/sd/5_0_0/device-discovery_sd.pdf). For further details about how this service is meant to be used, please consult that document.
 
-TODO
+## Interface Description
 
 ### register
 
-TODO
+The service operation **request** requires an authorization bearer header and a [DeviceRegistrationRequest](../data-models/device-registration-request.md)
+JSON encoded body.
+
+```
+POST /serviceregistry/device-registry/register HTTP/1.1
+Authorization: Bearer <authorization-info>
+
+{
+   "name":"thermometer2",
+   "metadata":{
+      "scales":[
+         "Kelvin",
+         "Celsius"
+      ],
+      "max-temperature":{
+         "Kelvin":310,
+         "Celsius":40
+      },
+      "min-temperature":{
+         "Kelvin":260,
+         "Celsius":-10
+      }
+   },
+   "addresses":[      
+      "address":"81:ef:1a:44:7a:f5"
+   ]
+}
+```
+
+The service operation **responses** with the status code `200 Ok` if called successfully and the device
+entity is already existing or `201 Create` if the entity was newly created. The response also contains a
+[DeviceRegistrationResponse](../data-models/device-registration-response.md) JSON encoded body.
+
+```
+{
+   "name":"thermometer2",
+   "metadata":{
+      "scales":[
+         "Kelvin",
+         "Celsius"
+      ],
+      "max-temperature":{
+         "Kelvin":310,
+         "Celsius":40
+      },
+      "min-temperature":{
+         "Kelvin":260,
+         "Celsius":-10
+      }
+   },
+   "addresses":[
+      {
+         "type":"MAC",
+         "address":"81:ef:1a:44:7a:f5"
+      }
+   ],
+   "createdAt":"2024-11-04T01:53:02Z",
+   "updatedAt":"2024-11-04T01:53:02Z"
+}
+```
+The **error codes** are, `400 Bad Request` if request is malformed, `403 Forbidden` if requester au-
```

Comment:
It is confusing, but 401 is for authentication and 403 is for authorization problems